X86/X86MachineInstructionRaiser.cpp

```diff
@@ -1207,6 +1207,34 @@ bool X86MachineInstructionRaiser::raiseBinaryOpRegToRegMachineInstr(
       raisedValues->setPhysRegSSAValue(dstReg, MBBNo, dstValue);
     }
   } break;
+  case X86::POPCNT16rr:
+  case X86::POPCNT32rr:
+  case X86::POPCNT64rr: {
+    const MachineOperand &DestOp = MI.getOperand(DestOpIndex);
+    assert(DestOp.isReg() && "Expecting destination of popcnt instruction to "
+                             "be a register operand");
+    assert((MCID.getNumDefs() == 1) &&
+           MCID.hasImplicitDefOfPhysReg(X86::EFLAGS) &&
+           "Unexpected defines in a popcnt instruction");
+    dstReg = DestOp.getReg();
+    Value *Src1Value = ExplicitSrcValues.at(0);
+
+    Module *M = MR->getModule();
+    Function *IntrinsicFunc =
+      Intrinsic::getDeclaration(M, Intrinsic::ctpop, Src1Value->getType());
+    Value *IntrinsicCallArgs[] = {Src1Value};
+    Instruction *BinOpInst =
+      CallInst::Create(IntrinsicFunc, ArrayRef<Value *>(IntrinsicCallArgs));
+
+    // No EFLAGS are effected
```

Comment:
Actually per https://www.felixcloutier.com/x86/popcnt
"Flags Affected
OF, SF, ZF, AF, CF, PF are all cleared. ZF is set if SRC = 0, otherwise ZF is cleared."

Would you be able to add code to abstract the specified eflags change?

Reply:
Thanks for pointing this out. I'll work on this.

Reply:
I've updated the code to reflect these changes to the eflags.